src/lib/core/option/option.ts

```diff
@@ -273,4 +273,26 @@ export class MatOption implements AfterViewChecked {
     return 0;
   }
 
+  /**
+   * Determines the position to which to scroll a panel in order for an option to be into view.
+   * @param optionIndex Index of the option to be scrolled into the view.
+   * @param optionHeight Height of the options.
+   * @param currentScrollPosition Current scroll position of the panel.
+   * @param panelHeight Height of the panel.
+   */
+  static getScrollPosition(optionIndex: number, optionHeight: number, currentScrollPosition: number,
```

Comment:
Could this be a standalone function rather than living on `MatOption`? Should this also be internal?

Reply:
I went for consistency since the `countGroupLabelsBeforeOption` is also a static method. I'm fine with making it internal, but I can see it being useful if somebody is implementing a component that uses `MatOption`.

Reply:
I'd rather keep the public API smaller and see if it comes up as a request first

Reply:
Done.